File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;
class Solution {
    public int solution(int[] priorities, int location) {        
        Queue<List<Integer>> queue = new LinkedList<>();
        
        PriorityQueue<Integer> heap = new PriorityQueue<>(Collections.reverseOrder());
        
        for(int i=0; i<priorities.length; i++) {
            Integer n = Integer.valueOf(priorities[i]);
            queue.offer(Arrays.asList(n, i));
            heap.offer(n);
        }
        
        int cnt=0;
        while(!queue.isEmpty()){
            List<Integer> top = queue.poll();
            if(top.get(0) == heap.peek()){
                cnt++;
                heap.poll();
                if(top.get(1) == location) break;
            }
            else queue.offer(top);
        }
        
        return cnt;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var answer = 0;
    let queue=[];
    for(let i=0; i<priorities.length; i++){
        queue.push([priorities[i], i]);
    }
    let priority=priorities.sort((a,b)=>b-a);
    while(queue.length>0){
        let [p, index]=queue.shift();
        if(priority[0]===p){
            priority.shift();
            answer++
            if(index===location) break;
        }
        else queue.push([p, index]);
    }
    return answer;
}